query-engine/query-engine-node-api/src/engine.rs

```diff
@@ -180,14 +180,16 @@ impl QueryEngine {
     }
 
     fn logger(&self) -> CallbackLogger {
+        // is a sql query?
```

Comment:
I don't understand what this is supposed to achieve.

Reply:
I am pretty sure I already explained this but I will do it again. _Right now_ we need to attach the logger to the future created by each of the operations (this will be removed after https://github.com/prisma/prisma-engines/pull/2477) the problem is, _right now_ you cannot use a single subscriber for all the futures (again, this will change soon) so you have to create it for each operation. That `logger` is pretty much a subscriber for the events generated for our logs (and very soon, traces and metrics).
Because _there are very events comming from the libs and our apps_ we have to be _selective_ about what events pass or not, if we enable the query logging, we need to pass those events (there is inconsistency in a lot of things in the libraries, for example, Quaint returns a lot of events under trace levels so we only need queries if requested, while mongo right now emits queries as `debug` levels -which is correct IMHO- and are much better described).
Given the previous statements, that function simply returns a subscriber with the correct filters attached to it and correctly process them as query or log events.